src/Storages/MeiliSearch/MeiliSearchConnection.cpp

```diff
@@ -0,0 +1,91 @@
+#include <string_view>
+#include <Storages/MeiliSearch/MeiliSearchConnection.h>
+#include <curl/curl.h>
+#include <Common/Exception.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int NETWORK_ERROR;
+}
+
+MeiliSearchConnection::MeiliSearchConnection(const MeiliConfig & conf) : config{conf}
+{
+}
+
+static size_t writeCallback(void * contents, size_t size, size_t nmemb, void * userp)
+{
+    (static_cast<std::string *>(userp))->append(static_cast<char *>(contents), size * nmemb);
+    return size * nmemb;
+}
+
+CURLcode MeiliSearchConnection::execQuery(std::string_view url, std::string_view post_fields, std::string & response_buffer) const
+{
+    CURLcode ret_code;
+    CURL * handle;
+    struct curl_slist * headers_list;
+
+    headers_list = nullptr;
+    headers_list = curl_slist_append(headers_list, "Content-Type: application/json");
+
+    String full_key;
+
+    if (!config.key.empty())
+    {
+        full_key = config.key_prefix + config.key;
+        headers_list = curl_slist_append(headers_list, full_key.c_str());
+    }
+
+    handle = curl_easy_init();
+    curl_easy_setopt(handle, CURLOPT_URL, url.data());
+    curl_easy_setopt(handle, CURLOPT_POSTFIELDS, post_fields.data());
+    curl_easy_setopt(handle, CURLOPT_POSTFIELDSIZE_LARGE, post_fields.size());
+    curl_easy_setopt(handle, CURLOPT_HTTPHEADER, headers_list);
+    curl_easy_setopt(handle, CURLOPT_MAXREDIRS, 50L);
+    curl_easy_setopt(handle, CURLOPT_CUSTOMREQUEST, "POST");
+    curl_easy_setopt(handle, CURLOPT_WRITEFUNCTION, writeCallback);
+    curl_easy_setopt(handle, CURLOPT_WRITEDATA, &response_buffer);
+
+    ret_code = curl_easy_perform(handle);
+
+    curl_easy_cleanup(handle);
```

Comment:
Is it exception free before this line of code? Or else we will have memory leak. I suggest using Poco http client instead of c-api.